### PR TITLE
feat: add post finder page as old search

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- We have now a dedicated page for the old search which can be found at `/posts/finder`.
- As mentioned in Slack, we will now keep the item in the sidebar originally intended for search but the name will vary based on which version you are at.
- For extension users with AI search, we will redirect to the new page when the sidebar item is clicked.
- Since we still have the legacy Search, we will reuse the same component but refactor all the words from "Search" to "Post finder" as per Slack message.
- Have also tested on extension with legacy Search to not redirect and work as it currently is instead.

This is deployed in [preview2](https://preview2.app.daily.dev/posts/finder).

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1871 #done

### CODE REVIEW

File: packages/shared/src/components/sidebar/DiscoverSection.tsx [hunk 1]
```diff
@@ -61,19 +62,16 @@ export function DiscoverSection({
...
-
-  if (searchValue === SearchExperiment.Control) {
-    discoverMenuItems.push({
+    {
       icon: (active: boolean) => (
         <ListIcon Icon={() => <SearchIcon secondary={active} />} />
       ),
-      title: 'Search',
+      title: isControlSearch ? 'Search' : 'Post finder',
       hideOnMobile: true,
```
Comment (rebelchris) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:70]: We can remove this if not control search

File: packages/extension/src/newtab/MainFeedPage.tsx [hunk 2]
```diff
@@ -8,6 +8,10 @@ import FeedLayout from '@dailydotdev/shared/src/components/FeedLayout';
 import dynamic from 'next/dynamic';
 import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
 import AlertContext from '@dailydotdev/shared/src/contexts/AlertContext';
+import { useFeature } from '@dailydotdev/shared/src/components/GrowthBookProvider';
```
Comment (rebelchris) [inline packages/extension/src/newtab/MainFeedPage.tsx:11]: Would this impact enrollment time for extension users? (When they first get enrolled)

**Review by rebelchris — COMMENTED** [review]
Early feedback, will ceck again tomorrow

File: packages/extension/src/newtab/MainFeedPage.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/extension/src/newtab/MainFeedPage.tsx:11]: The current process is Boot -> MainLayout -> this file. So there should be no impact.

File: packages/shared/src/components/sidebar/DiscoverSection.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:70]: Oh, right. Yes, will make that based on experiment.

Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:70]: Ah, right. Let me update it quickly.

**Review by rebelchris — APPROVED** [review]
Alright looks good now 👌

**Pull request merged**